File: src/services/TP.Condutores.Domain/Veiculo.cs
﻿using System;
using TP.Core.DomainObjects;

namespace TP.Condutores.Domain
{
    public class Veiculo : Entity
    {
        public Guid CondutorId { get; private set; }
        public string VeiculoId { get; private set; }
        public string Placa { get; private set; }
        public Condutor Condutor { get; private set; }

        public Veiculo(Guid idCondutor, string idVeiculo, string placa)
        {
            CondutorId = idCondutor;
            VeiculoId = idVeiculo;
            Placa = placa;
        }

        // EF Rel.
        protected Veiculo() { }
    }
}

File: src/web apps/Web/src/app/Telas/telas-principais/veiculo/models/veiculo.ts
export class Veiculo
{
    public id: string;
    public placa: string;
    public modelo: string;
    public marca: string;
    public cor: string;
    public anoFabricacao: number;
    public condutores: CondutorVeiculo[] = []
}    

class CondutorVeiculo
{
    public veiculoId: string;
    public condutorId: string;
    public cpf: string;
}